packages/json-encoder-web/index.ts

```diff
@@ -0,0 +1,47 @@
+import * as wasm from './pkg';
+import * as copy from 'copy-to-clipboard';
+// https://github.com/microsoft/monaco-editor/issues/2874
+import { editor as monacoEditor } from 'monaco-editor/esm/vs/editor/editor.api'
+
+self.MonacoEnvironment = {
+	getWorkerUrl: function (moduleId, label) {
+		return './json.worker.bundle.js';
+	}
+};
+
+const samplePayload = `[
+	{
+		"add": [
+			"A"
+		],
+		"message": "RegisterNetworks",
+		"remove": []
+	}
+]
+`;
+
+var editor = monacoEditor.create(document.getElementById('container'), {
+	value: samplePayload,
+	language: 'json',
+	minimap: {
+		enabled: false
+	},
+	theme: 'vs-light'
+});
+
+document.getElementById('compile-button').onclick = function () {
+	console.log('button was clicked');
+	let input = editor.getValue();
+	let compiled = wasm.compile(input, true);
```

Comment:
Currently, the form produces the output for `Calldata`, regardless of which option is selected.

I believe the fix revolves around this `true` value, considering whether the `Calldata` or `Bytes payload` option is selected instead.

Reply:
Right, I forgot about that missing piece. Fixed in https://github.com/graphprotocol/block-oracle/pull/205/commits/d5f225e1fcb6287ae9129e596a5ba1cb6e324744.